packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -422,6 +423,16 @@ function registerOrgPickerCommands(orgList: OrgList): vscode.Disposable {
   return vscode.Disposable.from(forceSetDefaultOrgCmd);
 }
 
+async function checkSObjectsRefreshed() {
+  const sobjectsCache = '.sfdx/typings/lwc/sobjects/**.d.ts';
+  const files = await vscode.workspace.findFiles(sobjectsCache);
+  if (!files || files.length === 0) {
+    const message = nls.localize('sobjects_refresh');
+    notificationService.showErrorMessage(message);
```

Comment:
Does it make sense to have this as an info message instead of an error? It's more of a friendly reminder than a problem happening.

Reply:
Might be nice to have a button the user can easily take action with. You can pass in a string as another argument for a button and the result returned from the call will be the item the user clicked, if they did at all.

Reply:
That's a good suggestion. Moving this check to apex will definitely help with this since the code for refresh is there and also easy to deal the case for users who have sobject refresh flag turned on, we don't do this check. Thanks!